### ISSUE TITLE
Multi-line arrow-function expression body should maintain absence or presence of line break after arrow.

### ISSUE THREAD

**Issue by WearyMonkey** [dprint/dprint-plugin-typescript#107]
**Describe the bug**

dprint-plugin-typescript version: 0.40.1

**Input Code**

```ts
colors.map(color => darken({
    color,
    opacity: 1
}));
```

**Expected Output**

```ts
colors.map(color => darken({
    color,
    opacity: 1
}));
```

**Actual Output**

```ts
colors.map(color =>
  darken({
    color,
    opacity: 1,
  })
);
```

[Playground link](https://dprint.dev/playground/#code/MYewNiBODOB0C2BDADgClBSACAvAPiwBNFIBrAUwDtUBvAKC0awygBoGmRlFgBLAFwCeALiwBGOgF8AlNIDcdIA/config/N4KABGBEA2CWB2BTA6rAJgFwBaQFxgA4AGAGnCgTUXg1Ux3wCYyJIBXAZ0QBUBDAIw54wAM17QuLKFwC2sAMIB7aIvhD8kAA4AnRCMTbIUyAEc2ijIgDKGAJ7REwyOIDuvWxwAiitvwdHySCQXABkERABpSidoEQDWTkQAIW1eAGNEdSgXLGoAOQsrBABzBzCkeKh+VIyABUUOWAxYVScOXhlEcsQAVXgHDg4ACV54YpLKyEaxhyTFNFt6xubWjXbO7smkAA8MJRptZQAxFRclppb4No6u8MmMVNg4MaUZGV4syFVoWwBZNmgzU2xkUmgMvAwim05xWVzWN2BgR0egMIzGE3wYgkiGMyP02iKM1uFUx4kkgV42kOLiObHgaVhADpErVKdRsJlMk53ggMLwEJN+AhKbYAKLbZGDS6M56IWoGcWSxqrURknGBToyfgKiW6KWqGXheXaRV65Vw1XY4x2MFhSypaCMriaSkQqFReBoNqIORKFRXYzUNgyTyINLQV1MzXagkutIYqA8mj8gOBIXwEWmzLmp1x6xsKk+T0lJJNFywLgAQU9le0TSwnWaaQA8mDUpDDPgHmx1aw0oo3uzuoyRFCMlY85WRParBGOLkst3e1B+2puwyobn0sk9FC5WyaAuuaSrYFVxx1x3uLYwVuMlP7XlEC4IohbC4oV6T+S+6oL9o2AZIpigzDAC0QO9EAfAwnxfN8P20L9LR-KA0EUZAsCeaw+UsToaEg6DtAwrDX3fT9hCXQMJShDBQ3DSNpQ4PMrALQ46TQEo8huNBFRoxcAOXSBR20ABJeAbAhH12QI6cDCOd14PIrsBOMYTmxECTcOkpjt0I+TtFIhCkMowJhM0qT8J0+9ZO0fTDKUsATNYMycIsjAZJnH0FGUP8KJU0y6QZS46IjdtGLzJJd10VldEPTksixFChMC2Es31eBIMi4T91ijkuAStVVJSy40pzKyoJs2l6VhezEOERLBOKRAMErNIMkGTdyqyvcYvZI8CtPVhYA01y8Pc8rCJEkRauM-yhpkTQaJChiDXKljC3YzjuJEha+L8ntjAAKw4bZSsufY+XCbRIPWtjizGM6VQa4xGyweZMqinK+vi+rCsCLgWra7NOoiz7eri-LfsGqA+WKJq0G4H1NAjSwPuyu1wWgfbBJtKD4HgCwIXC7dut0P0nr+1hccrQYDCZLrPsei0nOyTCHHMsaPIMYiHBm7HjFgEC9wKKhXjGxHdicNAdF5ABaQWCd0SYFb3I4sLF9kJYwKWZZoeWhd0WWRCwyAQAAXxAIA/language/typescript)

**Comment by dsherret** [comment]
Another similar example from @declanvong:

https://dprint.dev/playground/#code/GYexAoDsFMHcAIDyAjAVuA3gKHr+BDALnlBABoc9ljl8AnCgXwEpmBuIA/language/typescript

```ts
foo(new Obj({
    a: foo,
    b: bar,
}));
```

Outputs:

```ts
foo(
    new Obj({
        a: foo,
        b: bar,
    }),
);
```

**Comment by stagas** [comment]
I agree with @WearyMonkey, arrow functions that return single expression should be able to be compacted as much as they can, the current behavior discourages the usage of higher order functions in cases like currying etc.:

```js
const add = x => y => {
  return x + y
}
```

becomes:

```js
const add = x =>
    y => {
        return x + y;
    }
```
which is less readable as the "y" is now in a different line than the "x" so harder to reason about.

**Comment by tennox** [comment]
my example:
```typescript
const render = () => html`
    <div b="1 rd-4 gray-300"> 
        <button m-4 p-2 rd onclick="${addKid}">+</button>
    </div>
`
```
(`html` shouldn't go on next line)

Did not find any config to change this behaviour
